llvm/lib/Target/X86/X86InstrFragmentsSIMD.td

```diff
@@ -821,6 +821,88 @@ def X86vpdpbuuds : SDNode<"X86ISD::VPDPBUUDS", SDTVnni>;
 
 def X86Vmpsadbw : SDNode<"X86ISD::MPSADBW", SDTX86PSADBW>;
 
+def SDTAVX10SATCVT_BF162I : SDTypeProfile<1, 1, [
+  SDTCVecEltisVT<0, i16>, SDTCVecEltisVT<1, bf16>
+]>;
+
+def SDTAVX10SATCVT_PH2I : SDTypeProfile<1, 1, [
+  SDTCVecEltisVT<0, i16>, SDTCVecEltisVT<1, f16>
+]>;
+
+def SDTAVX10SATCVT_PS2I : SDTypeProfile<1, 1, [
+  SDTCVecEltisVT<0, i32>, SDTCVecEltisVT<1, f32>
+]>;
```

Comment:
Can we reuse `SDTFloatToInt`?

Reply:
VCVTNEBF162IBSZrr: 	(X86vcvtnebf162ibs:{ *:[v32i16] } VR512:{ *:[v32f16 v32bf16 v16f32 v8f64] }:$src)\
error: In VCVTNEBF162IBSZrr: Could not infer all types in pattern!

Reply:
You need to add `SrcInfo.VT` before `SrcInfo.RC:$src`, i.e., `(MaskNode (SrcInfo.VT SrcInfo.RC:$src))`

Reply:
aa5cc99